web3/wallets.js

```diff
@@ -505,6 +506,51 @@ exports.getAllWallets = async (req, res) => {
     } else if (Object.keys(res).length !== 0)
         return res.status(401).end('Account not found')
 }
+const getNetworkByToken = async (idCrypto) => {
+
+    try{
+        if (
+            cache.get('networks') &&
+            Date.now() - new Date(cache.get('networks')?.date).getTime() < 604800000
+        ) {
+            return cache.get('networks').data
+        }else{
+    
+        const options = {
+            method: 'GET',
+            url: process.env.CMC_CRYPTO_DETAILS,
+            params: {
+                id: idCrypto
+            },
+            headers: {
+                'X-CMC_PRO_API_KEY': process.env.CMCAPIKEY,
+            },
+           
+        }
+       
+        var results 
+        try {
+            results = await Promise.all([ rp.request(options)])
+
+        } catch (error) {
+            throw new Error('Error fetching  networks')
+        }
+   let networksContract = []
+     const  result = await results[0].data.data
```

Comment:
Consider removing await in this line (539). As I can see from the previous recap of the data at line 533, it seems unnecessary here. The Promise.all is already handling the promise resolution.